build/Dockerfile

```
@@ -172,28 +266,38 @@ apt-get install -y --no-install-recommends \
     curl \
     libpq5 \
     redis \
+    libboost-log1.74.0 \
     redis-tools
 rm -rf /var/lib/apt/lists/*
 EOF
 
-# Copy Rust binaries
-ARG BASE_PATH
-WORKDIR ${BASE_PATH}/bin
-ARG RUST_BUILD_PATH
-COPY --from=rust-builder ${RUST_BUILD_PATH}/target/release/cartesi-rollups-* .
-ENV PATH="${BASE_PATH}/bin:${PATH}"
+# Copy server-manager.
+COPY --from=server-manager /usr/bin/server-manager /usr/bin
 
-# Copy Go binary
-ARG GO_BIN_PATH
-COPY --from=go-builder ${GO_BIN_PATH}/cartesi-rollups-node .
+# Copy Rust binaries.
+# Explicitly copy each binary to avoid adding unnecessary files to the runtime image.
+ARG RUST_BUILD_PATH
+ARG RUST_TARGET=${RUST_BUILD_PATH}/target/release
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-advance-runner /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-authority-claimer /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-dispatcher /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-graphql-server /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-host-runner /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-indexer /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-inspect-server /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-state-server /usr/bin
+
+# Copy Go binary.
+ARG GO_BUILD_PATH
+COPY --from=go-builder ${GO_BUILD_PATH}/cartesi-rollups-node /usr/bin
 
-# Setup runtime dir
-ARG RUNTIME_DIR
-RUN <<EOF
-mkdir -p ${RUNTIME_DIR}
-chown cartesi:cartesi ${RUNTIME_DIR}
-EOF
+# Setup runtime dir.
+ARG RUNTIME_DIR=/usr/share/rollups-node
 WORKDIR ${RUNTIME_DIR}
+RUN chown cartesi:cartesi ${RUNTIME_DIR}
 
+# Set user to low-privilege.
 USER cartesi
-CMD ["cartesi-rollups-node"]
+
+# Set the Go supervisor as the command.
+CMD cartesi-rollups-node
```

This way `PID 1` will be sinde `sh -c`.

I'm assuming the go-supervisor is traping the signals, so it's better to use the exec form of `CMD`

```suggestion
CMD [ "cartesi-rollups-node" ]
```

See: https://docs.docker.com/engine/reference/builder/#shell-and-exec-form